demes/demes.py

```diff
@@ -1997,6 +1997,36 @@ def in_generations(self) -> Graph:
         graph.generation_time = 1
         return graph
 
+    def rename_demes(self, names: MutableMapping[str, str]) -> Graph:
```

Comment:
What's the reasoning for this being a `MutableMapping` rather than just a `Mapping`? This function shouldn't be modifying the mapping, right?

Reply:
The reasoning for the `MutableMapping` was for the scenario where you only want to change the label of a single deme  but not mess with all of the other demes (specifically the ancestral demes). The basis is that if an existing deme name does not exist you may want to default to keeping that name by adding it to the renaming mapping (see code below). 

https://github.com/aabiddanda/demes-python/blob/e2f6eb1fd78dab44045daa080add5f17bdcaf5d9/demes/demes.py#L2013-L2015


This relaxes the requirement that every deme needs to be defined in the names dictionary as well. If you think that adding in conditional statements to the cases to only rename demes that are named (rather than all demes in the graph) I can certainly implement that if its more in keeping with the spec.

Reply:
Yes, renaming only a subset of the demes seems reasonable, but I don't think we should be mutating the argument here. Using conditionals would be fine, but if you want to keep the code as it is, then you could copy the `names` dict at the start of the function.